save results as csv

aryanbhosale opened the pull request:
# Pull Request

## Description

Saves the result of the forecast as a csv file the moment it's run, will help us to plot the weekly accuracy

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

Sukh-P commented on examples/results/predictions_with_recent_pv_20240704_163005.csv:
```diff
@@ -0,0 +1,193 @@
+,power_kw,power_kw_no_live_pv
```
As per my comment via slack I would remove this file for now till we have more results

Sukh-P commented on examples/inverter_example.py:
```diff
@@ -26,6 +24,21 @@ def main():
 
     predictions_with_recent_pv_df["power_kw_no_live_pv"] = predictions_df["power_kw"]
 
+    # Get the directory of the current script
```
Nice that you have added the functionality to save to csv in, I would perhaps add a parameter and if statement in main to make this saving to csv optional? 

Then you could perhaps use the `typer` library https://typer.tiangolo.com/ to be able to set this parameter via the command line option e.g. if you wanted to save the outputs to csv you could run

`python examples/inverter_example.py --save-outputs=True`

aryanbhosale replied:
Yeah this sounds good, will do this next

aryanbhosale replied:
Hey @Sukh-P , I have made these changes and tested them, and i've also deleted that csv file from the repo for now, it's still stored locally, will merge it as one big csv file later when we have enough data

Sukh-P commented on examples/inverter_example.py:
```diff
@@ -1,15 +1,17 @@
 from quartz_solar_forecast.pydantic_models import PVSite
-from datetime import datetime, timezone
+import os
+import typer
```
Does typer need to be added to the requirements.txt file as well?

aryanbhosale replied:
> Does typer need to be added to the requirements.txt file as well?

Yup, updated

aryanbhosale commented on examples/inverter_example.py:
```diff
@@ -1,15 +1,17 @@
 pd.options.plotting.backend = "plotly"
 
-def main():
+app = typer.Typer()
```
> And then as long as the parameter has a type hint and default the `--save-outputs` parameter bit is automatically added in

Ive made these changes

Sukh-P replied:
Great, has this been tested locally again? Also just an FYI I think you could simplify the function signature bit still to this if you wanted to 
```
def main(save_outputs: bool = False)):
    .....

```

aryanbhosale replied:
> Great, has this been tested locally again? Also just an FYI I think you could simplify the function signature bit still to this if you wanted to
> 
> ```
> def main(save_outputs: bool = False)):
>     .....
> ```

Alright can shorten it

aryanbhosale replied:
> Also just an FYI I think you could simplify the function signature bit still to this if you wanted to
> 
> ```
> def main(save_outputs: bool = False)):
> ```

Hey Sukh, I have made these changes

Sukh-P commented:
@aryanbhosale is it alright if we split things up a bit more instead of adding on to this PR? I think it would be nice to have just the save to csv functionality bit in this PR and the streamlit frontend bits in another PR, thanks

aryanbhosale commented:
> @aryanbhosale is it alright if we split things up a bit more instead of adding on to this PR? I think it would be nice to have just the save to csv functionality bit in this PR and the streamlit frontend bits in another PR, thanks

Yeah that makes sense, how do I undo a commit to this pr?

aryanbhosale commented:
> @aryanbhosale is it alright if we split things up a bit more instead of adding on to this PR? I think it would be nice to have just the save to csv functionality bit in this PR and the streamlit frontend bits in another PR, thanks

I've removed the frontend in my latest commit

Sukh-P commented on requirements.txt:
```diff
@@ -8,4 +8,7 @@ openmeteo-requests==1.2.0
-xgboost==2.0.3
\ No newline at end of file
+xgboost==2.0.3
+plotly
```
Sorry small thing but can you remove `plotly` and`streamlit` for now? They were just needed for the frontend bit right so can add them in later?

aryanbhosale replied:
> Sorry small thing but can you remove `plotly` and`streamlit` for now? They were just needed for the frontend bit right so can add them in later?

Plotly is needed for the graph even in the CLI version, so should i just remove streamlit for now?

aryanbhosale replied:
> Sorry small thing but can you remove `plotly` and`streamlit` for now? They were just needed for the frontend bit right so can add them in later?

removed streamlit
